flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -88,6 +88,13 @@
   private transient Table table;
   private transient long maxCommittedCheckpointId;
 
+  // There're two cases that we restore from flink checkpoints: the first case is restoring from snapshot created by the
+  // same flink job; another case is restoring from snapshot created by another different job. For the second case, we
+  // need to maintain the old flink job's id in flink state backend to find the max-committed-checkpoint-id when
+  // traversing iceberg table's snapshots.
+  private static final ListStateDescriptor<String> JOB_ID_DESCRIPTOR = new ListStateDescriptor<>(
```

Comment:
I am wondering if it is better to consolidate all committer checkpoint states into a single structure, e.g. Pojo class or Avro record which are Flink supported state types for schema evolution.

Reply:
We've considered to use avro or pojo to consolidate the job-id and data files into a single structure. It could be considered as two separate issue: 
1.  use avro or pojo to serialize/deserialize.  If use avro, then we need the detailed schema for the whole structure,  but the `DataFile`  hides its schema inside a non-public implementation `GenericDataFile`, that was designed intentionally because we don't want to expose the detail schema to upper users for iceberg-core.  POJO need all fields provide getter/setter, while the DataFile don't support setter now.  More discussion could be found. 
2. Making them into a single structure.  I think we could do but I'm not sure what's the benefit.

Reply:
fair enough. agree that we should expose the GenericDataFile.

On the other hand, this can be simpler if we decided to just checkpoint the manifest file. Then we only need to track the manifest file path along with other metadata. Here is our Avro schema file.

```
{
    "type": "record",
    "name": "ManifestFileState",
    "namespace": "com.netflix.spaas.nfflink.connector.iceberg.model",
    "fields": [
        {"name":"path", "type":"string"},
        {"name":"length", "type":"long"},
        {"name":"specId", "type":"int"},
        { "name":"checkpointId", "type": "long"},
        { "name":"checkpointTimestamp", "type": "long"},
        { "name":"dataFileCount", "type": "long"},
        { "name":"recordCount", "type": "long"},
        { "name":"byteCount", "type": "long"},
        { "name":"lowWatermark", "type": ["null", "long"], "default": null},
        { "name":"highWatermark", "type": ["null", "long"], "default": null}
    ]
}
```